contentcuration/contentcuration/static/js/edit_channel/models.js

```diff
@@ -1,63 +1,284 @@
+    	}else{
+    		title += " (Copy)";
+    	}
+	},
+	copy_children:function(node, original_collection){
+		var self = this;
+		var parent_id = node.id;
+		var copied_collection = new NodeCollection();
+		copied_collection.get_all_fetch(original_collection);
+		$(copied_collection.models).each(function(){
+			console.log("end",this.duplicate(parent_id));
+		});
+	},
+	validate:function (attrs, options){
+		var self = this;
+
+		console.log("Checking if title is blank...");
+		//Case: title blank
+		if(attrs.title == "")
+			return "Name is required.";
+		if(attrs.parent){
+			var parent = new NodeModel({'id': attrs.parent});
+			parent.fetch({async:false});
+			if(attrs.kind == "topic"){
+				console.log("Checking if topic is descendant of itself..");
+				//Case: is a child of itself
+				if(parent.id == self.id)
+					return "Cannot place topic under itself."
+
+				//Case: is a child of its descendants
+				var temp = new NodeModel({'id': parent.get("parent")});
+				while(temp.get("parent")){
+					temp = new NodeModel({'id': parent.get("parent")});
+					temp.fetch();
+					if(temp.id == self.id)
+						return "Cannot place topic under any of its subtopics."
+				}
+			}
+
+			console.log("Checking if title already exists in topic..");
+			//Case: topic with same name exists in children
+			if(!this.siblings)
+				this.siblings = new NodeCollection();
+			if(!this.parent_children || parent.get("children") != this.parent_children){
+				this.parent_children = parent.get("children");
+				this.siblings = this.siblings.get_all_fetch(this.parent_children);
+			}
+			for(var i = 0; i < this.siblings.models.length; i++){
+				if(this.siblings.models[i].get("title") == attrs.title && this.siblings.models[i].id != self.id){
+					return "Name already exists under this topic. Rename and try again.";
+				}
+			}
+		}
+		console.log("Validated!");
+	},
+});
+
+var NodeCollection = Backbone.Collection.extend({
+	model: NodeModel,
+	/*save: function() {
+		$(this.models).each(function(){
+			this.save();
+		});
+	},*/
+
+	url: function(){
+       return window.Urls["node-list"]();
+    },
+
+   /* TODO: would be better to fetch all values at once */
+    get_all_fetch: function(ids){
+    	console.log("Calling get_all_fetch on ", ids);
+    	var to_fetch = new NodeCollection();
+    	for(var i = 0; i < ids.length; i++){
+    		var model = this.get({id: ids[i]});
+    		if(!model){
+    			model = this.add({'id':ids[i]});
+    			model.fetch({async:false});
+    		}
+    		to_fetch.add(model);
+    	}
+    	console.log("get_all_fetch is", to_fetch);
+    	return to_fetch;
+    },
+    sort_by_order:function(){
+    	this.comparator = function(node){
+    		return node.get("sort_order");
+    	};
+    	this.sort();
+    },
 });
 
 var TopicTreeModel = Backbone.Model.extend({
+	get_root: function(){
+		var root = new NodeModel({id: this.get("root_node")});
+		root.fetch({async:false});
+		return root;
+	},
+	urlRoot: function() {
```

Comment:
If a `url` method is defined for the collection, there is no need to define it for the Model too.